http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/CacheMetrics.scala

```diff
@@ -0,0 +1,39 @@
+package org.broadinstitute.dsde.workbench.leonardo.util
+
+import cats.effect.{Effect, Timer}
+import cats.implicits._
+import com.google.common.cache.CacheStats
+import fs2.Stream
+import io.chrisdavenport.log4cats.Logger
+import org.broadinstitute.dsde.workbench.openTelemetry.OpenTelemetryMetrics
+
+import scala.concurrent.duration._
+
+class CacheMetrics[F[_]: Timer: Effect] private (name: String, sizeF: F[Long], statsF: F[CacheStats])(
+  implicit metrics: OpenTelemetryMetrics[F],
+  logger: Logger[F]
+) {
+  val process: Stream[F, Unit] =
```

Comment:
how about moving `sizeF` and `statsF` to arguments for `process`? and make both arguments a thunk (`() => F[???]`)?  Just so it's more obvious these 2 are functions that will be run on an interval?

Reply:
I could try that.. originally I had `CacheMetrics` as an object and just passed all params to `process`. Maybe that's actually cleaner.

Reply:
you probably won't be able to make it an object because of all the implicits required...another thing I would do is at call site, just do `def recordCacheMetricsProcess: Stream[F, Unit] = CallMetrics[F].process`